test/files/rules/nocondassign.test.ts

```diff
@@ -0,0 +1,34 @@
+// valid cases
+if (x == 5) { }
+if (x === 5) { }
+else if (y <= 23) { }
+else if ((z && a) == 7) { }
+else { }
+
+do { } while (x == 2);
+do { } while (x !== 2);
+
+while (x == 2) { }
+while (x !== 2) { }
+
+for (var x = 8; x == 8; ++x) { }
+for (var x = 8; x == 8; x = 12) { }
+for (;;) { }
+
+// invalid cases
+if (x = 5) { }
+if (a && (b = 5)) { }
+else if (x = 2) { }
```

Comment:
please add another test where there are two assignments in the conditional expression; the linter should report failures on both